### PR TITLE
[core,gateway] migrate rpc gateway to use new transport layer.

### PR DESCRIPTION
**Pull request by llyzs**

### CODE REVIEW

**Review by hardening — APPROVED** [review]
LGTM

File: libfreerdp/core/gateway/rpc.c [hunk 1]
```diff
@@ -707,10 +707,9 @@ static RpcVirtualConnection* rpc_virtual_connection_new(rdpRpc* rpc)
 
 static BOOL rpc_channel_tls_connect(RpcChannel* channel, UINT32 timeout)
 {
-	int sockfd = 0;
 	rdpTls* tls = NULL;
 	int tlsStatus = 0;
-	BIO* socketBio = NULL;
+	BIO* layerBio = NULL;
```
Comment (hardening) [inline libfreerdp/core/gateway/rpc.c:712]: perhaps this could be moved line 735 so that we don't have that unnecessary NULL initialisation

**Review by mfleisz — APPROVED** [review]
Tested and works when connecting to an RPC gateway 👍

File: libfreerdp/core/gateway/rpc.c (hunk 1 above)
Comment (llyzs) [reply libfreerdp/core/gateway/rpc.c]: While I was at it I also moved the other variables and add some asserts.

**Review by akallabeth — APPROVED** [review]
LGTM

**Comment by MartinHaimberger** [comment]
+1 LGTM

**Review by MartinHaimberger — APPROVED** [review]

**Pull request merged**